client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java

```diff
@@ -124,7 +125,8 @@ private static final class CelebornInputStreamImpl extends CelebornInputStream {
     private final CelebornConf conf;
     private final TransportClientFactory clientFactory;
     private final String shuffleKey;
-    private PartitionLocation[] locations;
+    private ArrayList<PartitionLocation> locations;
+    private ArrayList<PbStreamHandler> streamHandlers;
```

Comment:
Requiring the two objects to have the same size does not feel like a good idea. Should we consider using a pair so that even if some handler fails, you can still use those success handlers.